### PR TITLE
Improve monitoring facility

### PR DESCRIPTION
**Pull request by hossainemruz**
Fixes: #582, #585 
Monitoring:
- [x] Add in Script
- [x] Add in Helm
- [x] Update Docs
- [x] Update dashboard json file

@galexrt we have made some adjustment in grafana dashboard json file. Can you please update in official grafana repo or show us how we can update this?

### CODE REVIEW

**Comment by galexrt** [comment]
For some weird reason I currently cannot login to Grafana.com to update the dashboards there already.
I will check if I can login this evening again.

**Comment by galexrt** [comment]
I have updated the dashboard on Grafana.com https://grafana.com/dashboards/4198/revisions.

**Comment by tamalsaha** [comment]
Thanks @galexrt ! Is there anyway to give us permission to update the dashboard on Grafana.com ?

**Comment by galexrt** [comment]
@tamalsaha I just checked on grafana.com, sadly I can't add "members" to my (personal) "org" which is "galexrt".
I would need a "non-personal" org to add members to an org but for that I need to have a subscription active for some service of theirs it seems.

Maybe one of you could ask the Grafana.com support about an open source free org for the project/appscode org (maybe other appscode projects can make use of Grafana dashboards too in some way)?

File: chart/stash/templates/apiregistration.yaml [hunk 1]
```diff
@@ -58,6 +58,26 @@ data:
   tls.crt: {{ b64enc $cert.Cert }}
   tls.key: {{ b64enc $cert.Key }}
 ---
+{{ $promNamespace:= default .Release.Namespace .Values.monitoring.prometheusNamespace }}
+{{- if (and (ne $promNamespace .Release.Namespace) (eq .Values.monitoring.operator true)) }}
```
Comment (tamalsaha) [inline chart/stash/templates/apiregistration.yaml]: direct `and`

File: chart/stash/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    release: "{{ .Release.Name }}"
+spec:
+  namespaceSelector:
+    matchNames:
+    - {{ .Release.Namespace }}
+  selector:
+    matchLabels:
+      app: "{{ template "stash.name" . }}"
+  endpoints:
+  {{- if eq .Values.monitoring.backup true }}
```
Comment (tamalsaha) [inline chart/stash/templates/servicemonitor.yaml]: direct `if`

File: docs/examples/monitoring/coreos/prometheus.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+apiVersion: monitoring.coreos.com/v1
```
Comment (tamalsaha) [inline docs/examples/monitoring/coreos/prometheus.yaml]: Why not just use official instructions?

File: docs/examples/monitoring/coreos/prometheus-operator.yaml [hunk 4]
```diff
@@ -0,0 +1,109 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: prometheus-operator
```
Comment (tamalsaha) [inline docs/examples/monitoring/coreos/prometheus-operator.yaml]: Why not just use official instructions?

File: hack/deploy/monitor/servicemonitor-backup.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: stash-servicemonitor
+  namespace: ${PROMETHEUS_NAMESPACE}
+  labels:
```
Comment (tamalsaha) [inline hack/deploy/monitor/servicemonitor-backup.yaml]: Should be a flag.

File: chart/stash/templates/apiregistration.yaml (hunk 1 above)
Comment (hossainemruz) [reply chart/stash/templates/apiregistration.yaml]: fixed

File: chart/stash/templates/servicemonitor.yaml (hunk 2 above)
Comment (hossainemruz) [reply chart/stash/templates/servicemonitor.yaml]: fixed

File: docs/examples/monitoring/builtin/prom-config.yaml [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    scrape_configs:
+    - job_name: stash-pushgateway
+      scrape_interval: 30s
+      scrape_timeout: 10s
+      metrics_path: /metrics
+      scheme: http
+      honor_labels: true
+      kubernetes_sd_configs:
+      - role: endpoints
+      relabel_configs:
```
[the remark this answers is not in the document]
Comment (hossainemruz) [reply docs/examples/monitoring/builtin/prom-config.yaml]: fixed

File: hack/deploy/monitor/servicemonitor-backup.yaml (hunk 5 above)
Comment (hossainemruz) [reply hack/deploy/monitor/servicemonitor-backup.yaml]: used flag

File: docs/examples/monitoring/coreos/prometheus-operator.yaml (hunk 4 above)
Comment (hossainemruz) [reply docs/examples/monitoring/coreos/prometheus-operator.yaml]: used official instruction

File: docs/examples/monitoring/builtin/prom-config.yaml [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      - source_labels: [__meta_kubernetes_service_label_app]
```
[the remark this answers is not in the document]
Comment (hossainemruz) [reply docs/examples/monitoring/builtin/prom-config.yaml]: added comment to clarify it is default value and user should use label based on their setup.

**Pull request merged**